testing/file_test/file_test_base_test.cpp

```diff
@@ -62,32 +62,34 @@ class FileTestBaseTest : public FileTestBase {
       int i = 0;
       for (const auto& file : test_files) {
         // Prints line numbers to validate per-file.
-        stdout << file.filename << ":2: " << ++i << "\n";
+        stdout << file.filename << ":1: " << ++i << "\n";
       }
       return true;
+    } else if (filename == "alternating_files.carbon") {
+      stdout << "unattached message 1\n"
+             << "a.carbon:2: message 2\n"
+             << "b.carbon:5: message 3\n"
+             << "a.carbon:2: message 4\n"
+             << "b.carbon:5: message 5\n"
+             << "unattached message 6\n";
+      stderr << "unattached message 1\n"
+             << "a.carbon:2: message 2\n"
+             << "b.carbon:5: message 3\n"
+             << "a.carbon:2: message 4\n"
+             << "b.carbon:5: message 5\n"
+             << "unattached message 6\n";
+      return true;
+    } else if (filename == "unattached_multi_file.carbon") {
+      stdout << "unattached message 1\n"
+             << "unattached message 2\n";
+      stderr << "unattached message 3\n"
+             << "unattached message 4\n";
+      return true;
     } else {
       return ErrorBuilder() << "Unexpected file: " << filename;
     }
   }
 
-  auto ValidateRun(const llvm::SmallVector<TestFile>& test_files)
```

Comment:
@jonmeow It wasn't clear to me how much value this extra validation was adding, and it seemed to be getting a little unmanageable as I added more multi-file tests. Is this OK to remove?

Reply:
Just a note that PR #3183 moves its logic but continues to preserve the behavior.

Reply:
Thanks, I think I understand what you're looking for here. I've put the split results into the stdout for this test so that we can still validate the splitting.

Reply:
@jonmeow FYI, the code here changed a fair bit when merging in #3183, you might want to take another look to make sure you're happy before we merge.